python/ray/state.py

```diff
@@ -866,6 +866,15 @@ def nodes():
     return state.node_table()
 
 
+def workers():
+    """Get a list of the workers in the cluster.
```

Comment:
Please modify the docstring here. This is wrong. This only gets a list of workers that failed.

Reply:
Actually, it is not that the docstring is wrong, but our worker updating logic is not correct. That's why you also couldn't find proper worker info. If you look at this file
<img width="876" alt="Screen Shot 2020-10-26 at 12 19 49 AM" src="https://user-images.githubusercontent.com/18510752/97144119-2100eb00-1721-11eb-81b6-96a6e12f132f.png">

As you can see in the core_worker.cc, it updates worker info to GCS. But this is not returned by state.workers because of this line.

```
    def workers(self):
        """Get a dictionary mapping worker ID to worker information."""
        self._check_connected()

        # Get all data in worker table
        worker_table = self.global_state_accessor.get_worker_table()
        workers_data = {}
        for i in range(len(worker_table)):
            worker_table_data = gcs_utils.WorkerTableData.FromString(
                worker_table[i])
            # LOOK AT HERE. It requires is_alive to be true
            if worker_table_data.is_alive and \
                    worker_table_data.worker_type == gcs_utils.WORKER:
                worker_id = binary_to_hex(
                    worker_table_data.worker_address.worker_id)
                worker_info = worker_table_data.worker_info
```

The root cause is when a core worker updates its states, it doesn't update is_alive to be true. If you modify the core worker.cc file to have is_alive to be true, then you'll be able to see all those worker information by state.workers() method.

Reply:
get it~ thanks~